src/app_service/service/accounts/service.nim

```diff
@@ -295,14 +295,15 @@ QtObject:
       "profile-pictures-show-to": settings.PROFILE_PICTURES_SHOW_TO_EVERYONE,
       "profile-pictures-visibility": settings.PROFILE_PICTURES_VISIBILITY_EVERYONE,
       "url-unfurling-mode": int(settings.UrlUnfurlingMode.AlwaysAsk),
+      "omit-transfers-history-scan?": omitTransfersHistory,
```

Comment:
@saledjenic just to be sure: when we generate a new KK multiaccount, do we force user to write down and verify a mnemonic phrase? That was the case on the old version of mobile app and I don't know how to test it on desktop at the moment.

In case if we do show the phrase before account creation it is kind of not 100% safe to skip history scan (maybe 99.99999% safe but anyway), so I would just go with a regular history check with all KK accounts in that case. The point of this change in overall is to avoid making unnecessary requests during potential user acquisition, and adding it for KK multiaccounts will likely have almost no effect.  

But yes, if we don't force users to write down the phrase then no reason to not do this with KK multiaccs.

Reply:
Every newly created Status account, that the app is generating a seed phrase for, is forced to write that seed phrase down and to confirm, in a new step, a few randomly chosen words from it to prove that all is written well.

For such users, we don't need to perform any history checks, cause we're sure there are none, and that seed phrase has not been used ever before. And there are just 2 onboarding flows we don't need to check the history for:
1. creating a new Status account generating seed phrase on the statusgo side
2. creating a new Status account generating a seed phrase on the keycard

Reply:
Is that a part of onboarding flow? Or just a separate screen which can optionally be opened afterwards?

> For such users, we don't need to perform any history checks

If it is part of onboarding technically there is no guarantee that phrase was not used elsewhere before onboarding was finished. That is the point, and that's where it can break. So if you say that's the case (during onboarding) even for a regular accs that might not be optimal for other possible optimizations, like not checking history at all if phrase has not been shown. Would be nice to figure out what is the reasoning behind making it mandatory for a regular acc considering obvious cons...

Reply:
Yes, that's the step of the onboarding flow. 

But if I get you well, the only use case for what you're saying is:
- a user starts the app to create a brand-new Status account
- at the moment when the app displays a seed phrase user gets that seed phrase uses it in any other wallet, or other Status app instance, without closing this one, and makes some transactions for it using those other apps
- then get back to this onboarding process and complete it

But you have to admit that's a very uncommon use case and if anyone gets into it, we don't need to take care about that at all, cause that's cheating. :)

Also in the desktop app onboarding flow, if you seed the seed phrase and go back and then again go forth to see the seed phrase again, or close the app and run it again, it will be a new seed phrase.

This is the step I am talking about:
https://www.figma.com/file/17fc13UBFvInrLgNUKJJg5/Kuba%E2%8E%9CDesktop?type=design&node-id=13653-451025&mode=design&t=536aKelUHZR88aZq-0

Reply:
@saledjenic , per that figma link it looks exactly like I though in the first place: for a regular non-KK multiacc there is no mandatory step of saving 12-words mnemonic during onboarding, that's the case only for KK multiaccs.

I think my point is still valid: regardless of whether it is cheating or not, considering that the number of KK accounts is going to be almost negligible during any possible user acquisition campaign and this change addresses specifically unnecessary calls during that acquisition, there is no point in having that possibility of "cheating" in the app. We can just go with initial scan for all KK accounts, both generated or recovered.

Reply:
Could be that I don't get you well.

If I understand correctly, this PR is about skipping a history scan for a newly created Status account. That means we do not perform any history scan for the default wallet account address (that was generated at the moment of Status account creation) as well as for any newly created wallet account later, that is generated for the profile keypair.

@rasom could you confirm, is that correct what I said?

Reply:
> as well as for any newly created wallet account later, that is generated for the profile keypair.

That part is not the case at the moment. For that we would need to track whether the user has seen 12-words or/and synced multiacc, because both of those action might have some effect here. I'm working on this separately now, and it is not to address the initial scan but rather idle RPC load, because in some cases not only we don't need to make an initial scan but also watching for new transfers is unnecessary.

Anyway, currently it is only about the first wallet acc which is generated right away without any additional steps from the user.

Reply:
> Anyway, currently it is only about the first wallet acc which is generated right away without any additional steps from the user.

Ok, if it is only about the default wallet account, that is one with path `m/44'/60'/0'/0/0` and one that is created along with the Status chat account creation, then we have to do the same for creating it via Keycard as well. And that was the point in my first comment here.

Reply:
@saledjenic we don't need to do it for KK account because
1. The effect of it is going to be negligible (the number of KK multiaccs compared to regular ones is going to be low)
2. Doing this opens an extremely unlikely to be used (but still possible) way to ignore an account which actually has history.

So we gain (almost) nothing and open very tiny, nevertheless, possibility for a bug by adding this behavior for KK multiaccs. And I don't see a reason to open that possibility.

Reply:
@saledjenic `recoverAccount` is passed now